src/moveir/declaration.rs

```diff
@@ -36,7 +36,7 @@ pub(crate) struct MoveVariableDeclaration {
 }
 
 impl MoveVariableDeclaration {
-    pub fn generate(&self, function_context: &FunctionContext) -> MoveIRExpression {
+    pub fn generate(&self, function_context: &mut FunctionContext) -> MoveIRExpression {
```

Comment:
A lot of these (`fn generate`s in `moveir/*.rs` are now `&mut`, just want to check the reason behind this

Reply:
Good spot, this must have been the result of some failed experiment where mutable function context was required. I have removed all this so it is now minimally mutable